weave/trace_server/table_query_builder.py

```diff
@@ -39,16 +39,16 @@ def make_natural_sort_table_query(
     SELECT DISTINCT tr.digest, tr.val_dump, t.row_order
     FROM table_rows tr
     INNER JOIN (
-        SELECT row_digest, row_number() OVER () AS row_order
+        SELECT row_digest, row_number() OVER () AS row_order, rn
         FROM (
-            SELECT {row_digests_selection} as row_digests
+            SELECT {row_digests_selection} as row_digests, row_number() OVER (PARTITION BY project_id, digest) AS rn
```

Comment:
Since you don't actually care about ordering, i think you can just do `LIMIT 1` in the inner query in both cases and reduce the overall cost of this query